### ISSUE TITLE
Forwarded header does not comply with rfc7239 when using router v4v6/v6 mode

### ISSUE THREAD

**Issue by alexcern** [openshift/origin#20211]
Since the upgrade to HAProxy 1.8 in the Openshift 3.9 Haproxy router image, the `Forwarded` header is not compliant with rfc7239 anymore (IPv6 addresses should be double-quoted).

This breaks applications that parse the `Forwarded` header in order to obtain information about the initial client IP address and protocol.

##### Version

oc v3.9.0+ba7faec-1
kubernetes v1.9.1+a0ce1bc657
features: Basic-Auth GSSAPI Kerberos SPNEGO

##### Steps To Reproduce
1. Deploy router in `v4v6` or `v6` mode
2. Inspect the `Forwarded` header value

##### Current Result

`Forwarded` header looks like:

```
Forwarded: for=[::ffff:1.2.3.4];host=my.dns.domain;proto=https
```

##### Expected Result

As per RFC7239 the IPv6 address should be in double-quotes:

```
Forwarded: for="[::ffff:1.2.3.4]";host=my.dns.domain;proto=https
```

##### Additional Information

The double-quotes [are present in the haproxy config template](https://github.com/openshift/origin/blob/1cdf91738ccf94ddc94cd96d0cb9907e9de89f75/images/router/haproxy/conf/haproxy-config.template#L362) 
but due to [new escaping rules in HAProxy 1.8](http://cbonte.github.io/haproxy-dconv/1.8/configuration.html#2.2) they must now be escaped.

The same template was working fine with HAProxy 1.5 where escaping was not necessary.

**Comment by jwforres** [comment]
@openshift/sig-network-edge

**Comment by alexcern** [comment]
Fixed by #20213

/close
